src/Symfony/Component/Validator/Tests/Mapping/Factory/LazyLoadingMetadataFactoryTest.php

```diff
@@ -168,6 +168,22 @@ public function testMetadataCacheWithRuntimeConstraint()
 
         $metadata = $factory->getMetadataFor(self::CLASS_NAME);
     }
+
+    public function testGroupsFromParent()
+    {
+        $reader = new \Symfony\Component\Validator\Mapping\Loader\StaticMethodLoader();
+        $factory = new LazyLoadingMetadataFactory($reader);
+        $metadata = $factory->getMetadataFor('Symfony\Component\Validator\Tests\Fixtures\EntityStaticCarTurbo');
+        $classMetaData = $metadata->getPropertyMetadata('wheels');
+        $constraints = $classMetaData[0]->getConstraints();
```

Comment:
The culprit here is that you are making the assumption that `getPropertyMetadata()` of the `ClassMetadata` class will online return one `PropertyMetadata` instance (or that the one you need is the first entry in the returned collection). It's true that this was the case before #20793, but that IMO is can implementation detail you shouldn't rely on (the `RecursiveContextualValidator`, for example, iterates over the return property metadata). I'll see if we can nonetheless update the code to only return one instance, but as said that's not something I would rely on.